### ISSUE TITLE
Website - User Input Form to Collect Questions

### ISSUE THREAD

**Issue by JaeW** [mgolokhov/dodroid#15]
Could we get more people to contribute questions if we had a website whose input form collected the data to a json file.  Or does that detract from the app work too much?

**Comment by mgolokhov** [comment]
Any help would be nice.
Person can push json file here.
Or create an issue with questions answers and tags (but it's better to look in json first).

**Comment by JaeW** [comment]
You can pull info from a Google spreadsheet, yes?  Created a <a href="https://docs.google.com/forms/d/12t_k5dga6CPrpOeP0kb1L9JexUK8PyRJNTRHlU9t24I/viewform?usp=send_form">Google Form</a> to collect data.  Crude but efficient.

**Comment by mgolokhov** [comment]
Wow!

**Comment by JaeW** [comment]
Here's the link to the form spreadsheet data.  Please let me know if this will work; if so I promise to help and not take us down rabbit holes. :-\

**Comment by JaeW** [comment]
Crap.  Forgot to add the <a href="https://docs.google.com/spreadsheets/d/13bmt8pwh4x4GFTnoctxkxjKjsxDtYwwXbGS6ZEB-ik8/edit#gid=1307737117"link</a>.  Here it is.

**Comment by JaeW** [comment]
Was that a good or bad 'Wow!'?  Will this work?

**Comment by mgolokhov** [comment]
Awesome!

**Comment by mgolokhov** [comment]
+Tags?

**Comment by JaeW** [comment]
Sorry, don't understand +Tags.  Are you asking if the user can add additional tags?

**Comment by mgolokhov** [comment]
Would be nice to have this option.

**Comment by JaeW** [comment]
Do you want the users to be able to add additional tags?  I'm always a little hesitant to allow the masses to add stuff like that.  I could totally add you and joe as collaborators so anyone working on the backend could make adjustments.  Would that be enough?

**Comment by mgolokhov** [comment]
May be to add switcher "Advanced mode", so me can push questions as well.

**Comment by JaeW** [comment]
I truly apologize.  It's 5am here, and my brain is not working well and I'm confused.  Would the "Advanced mode" be fore the App?  Or for this form which simply does data/question collection?  And where would would we be pushing questions?  Oh wait!  Or are you asking if there is a way to automatically push a new question submission to the question database?

**Comment by mgolokhov** [comment]
I'm talking about Google Form - implement two modes User and Advanced User (can be a checkbox or another link).

**Comment by JaeW** [comment]
Hmmmmm.  Not sure about that.  Would need to do some research.

**Comment by mgolokhov** [comment]
My english grammar can be wrong, feel free to correct (I'll be grateful).

**Comment by JaeW** [comment]
It appears that kind of customization is possible.  <a href="https://support.google.com/docs/answer/2942256?hl=en">Customizing forms</a>

**Comment by JaeW** [comment]
Your English is actually very, very good.  I think my lack of understanding has more to do with my newbie status.

**Comment by mgolokhov** [comment]
It's already cool. I like the visualization in tables.
About export in json
http://blog.pamelafox.org/2013/06/exporting-google-spreadsheet-as-json.html

**Comment by JaeW** [comment]
Oh excellent!   I agree the visualization in tables is much easier.  Especially if we want to check the goodness of the question/answer before importing it into the app.

**Comment by JaeW** [comment]
One thing I would (humbly) recommend, though.  Before we starting adding permissions, etc. to forms, let's see how many contributors we attract.  Since your initial post has only attracted you, me and Joe, I would like to come up with a "pitch" that will convince others that contributing to this project will benefit them.

**Comment by mgolokhov** [comment]
It's not an easy thing...
We can cross reference to the problems students get in labs and problems we get while coding (I do my best to comment code).
We can reference to the set of questions we added for a specific topic.
We can contact to teachers for an assistance (need solid app).

**Comment by mgolokhov** [comment]
By the way Dr. Douglas C. Schmidt from https://class.coursera.org/posaconcurrency-003
seems more active mentor.

**Comment by JaeW** [comment]
In your country's universities, do they have wash-out classes?  They do that at American universities.  Basically the first one or two classes in any course subject they give you very little support to see who has the determination to get through.  The further one gets through a course of study the more attention they give you.  I just assume that this first (and possibly second) class are just that - meant to weed people out.  

You're so right.  Getting most people to do anything outside of what is absoluttely required is difficult.  But the people who DO take that initiative are the ones who succeed.  I'm thinking that, if people could use their contribution to this project on a resume they might be more inclined to help.  Need to sleep on it though.  Human motivation/nature is the most complicated thing on earth.

**Comment by JayBrizle** [comment]
Yeah professor Schmidt is very active his courses. He's always on the forums and has live office hours for questions and everything. In the Android concurrency course he evens discusses a post in the forum he thought was good during one of his question and answer sessions. He walks through the assignment code and everything. I will say that class is much more technical though and that sort of thing is almost needed to make a class like that more similar to a real university online class(I believe that is their goal)

**Comment by mgolokhov** [comment]
Even three of us by pushing just 1-2 questions a day in a month we'll get more than 100 questions.

**Comment by mgolokhov** [comment]
Janice, can you make Question Hint same size as Answers fild?

**Comment by JayBrizle** [comment]
Yeah like Maxim said, coming up with questions won't be a problem with 3 of us. We will have a massive amount of those soon enough. The thing will just be deciding how to organize them, like we are starting to discuss now. This should be fun, and sometimes frustrating :/

**Comment by JaeW** [comment]
I changed the Hint size on the form, but I'm going to play devil's advocate again (sorry, but that's what I do).  When you import these fields into a database it will use the field type to allocate memory, yes?  So, by using the Google forms paragraph field type (which the hint field is now), even if the hint itself is only a few words I'm pretty sure that the DB will still allocate that larger memory size.  Depending on how many questions we end up with, this could really blow up the DB size.  Just a thought, but something to keep in mind as the larger the db gets the slower it processes.

**Comment by JaeW** [comment]
I like the idea (for now) of just the 3 of us adding questions.  I think I will add an additional field to collect our respective initials.  That way we can test each others' questions (i.e., questions I write will be tested by Joe and Maxim to make sure they make sense and there is a correct answer).  Okay with you guys?

**Comment by JaeW** [comment]
Suggestion for adding/contributing questions.  So we have a pretty equal number of questions for each tag/topic, what if we choose 3 or 4 tags each week.  Each of us agree to contribute at least 3 questions for each tag.  That way we have the advantage of different question styles.  Yes/no/maybe?

**Comment by mgolokhov** [comment]
Premature optimization is the root of all evil -- DonaldKnuth
And for text it's not a big deal.

**Comment by JaeW** [comment]
Point taken.  I'm learning.  It's painful - as it should be.

**Comment by JaeW** [comment]
That which does not kill us makes us stronger. -- Friedrich Nietzsche

**Comment by mgolokhov** [comment]
Sure we'll review each other.
As for equal number of tags it's a bit harder, because mostly we study topics sequentially.
Just add all question you like, we'll balance them later.

**Comment by mgolokhov** [comment]
Would be nice to have such feature:
when user enters a new question Form checks if similar already exists and show them.

**Comment by JaeW** [comment]
Maxim - nice job on the Google sheets/forms data validation.

**Comment by mgolokhov** [comment]
It is preliminary, validation rule is incomplete(we have to add all tags).

**Comment by mgolokhov** [comment]
Added python [script to export quiz table](https://gist.github.com/mgolokhov/639da24aa5623263d4f7#file-export_quiz_to_json-py)

**Comment by mgolokhov** [comment]
I integrated export script in the build system.
We hit 50 questions!
Big thanks to Janice!

**Comment by JaeW** [comment]
Very nice work on the Python script!  
Coming up with the form was easy.  You guys have done the hard (coding) work.  Let's just say kudos to all.  
Don't know about Joe, but I have to do some serious work on the final two course labs.   Need to put dodroid on back burner for about a week.  Totally available for questions, but might not be fast in responding.  Please forgive in advance.  :-)

**Comment by JaeW** [comment]
You know?  Lab4 actually wasn't that bad; I knocked that out last night.  The Modern Art lab is going to be interesting . . . . . looking . . . since I'm color blind.  Ha!  Don't think it'll take that long, though.  Like you said, Maxim, the layouts are witchy.  }:-)

**Comment by JayBrizle** [comment]
Whoa I forgot about the modern art thing. I've been locked down finishing the final labs for all 3 Android courses. I'm also taking other java courses and I'm in "real" school for software dev right now. I will get back on dodroid after the weekend when all these classes end Sunday night

**Comment by JaeW** [comment]
Holy crap!  You spend so much time programming you might as well have your
computer implanted.  :-D

No worries.  I think we all have deadlines coming up.  If I knock this
Modern Art thing out tonight I think I'm going to look more at Android's
Material Design guidelines.  Following latest Android guidelines will give
it the professional appearance I think we all want it to have.

Good luck with all of your assignments!  TTYL

On Thu, Nov 12, 2015 at 4:37 PM, Joe Brown notifications@github.com wrote:

> Whoa I forgot about the modern art thing. I've been locked down finishing
> the final labs for all 3 Android courses. I'm also taking other java
> courses and I'm in "real" school for software dev right now. I will get
> back on dodroid after the weekend when all these classes end Sunday night
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mgolokhov/dodroid/issues/15#issuecomment-156259442.

## 

Janice Wildrick
314-651-4597

“Never doubt that a small group of thoughtful, committed, citizens can
change the world. Indeed, it is the only thing that ever has.”
― Margaret Mead http://www.goodreads.com/author/show/61107.Margaret_Mead

**Comment by mgolokhov** [comment]
https://github.com/mgolokhov/dodroid/wiki/User-Input-Form-to-Collect-Questions
